server/tests/util/graphql-test-utilities.js

```diff
@@ -0,0 +1,50 @@
+const { ApolloServer } = require('apollo-server');
+const { createTestClient } = require('apollo-server-testing');
+const typeDefs = require('../../graphql-schema');
+const getGraphQLContext = require('../../graphql-context');
+const resolvers = require('../../resolvers');
+const defaultUser = require('../mock-data/newUser.json');
+
+let mockReq;
+
+const server = new ApolloServer({
+    typeDefs,
+    context: () => getGraphQLContext({ req: mockReq }),
+    resolvers
+});
+
+const { query: testClientQuery } = createTestClient(server);
+
+const failWithErrors = errors => {
+    let formatted = '';
+    errors.forEach(error => {
+        formatted +=
+            `GraphQL error in ${JSON.stringify(error.path)}:\n\n` +
+            `${error.message}\n\n` +
+            `(${JSON.stringify({
+                extensions: error.extensions,
+                location: error.location
+            })})\n`;
+    });
+    throw new Error(formatted);
+};
+
+/**
+ * Returns the data from a given query, useful for situations where errors are
+ * not expected and no query variables are needed.
+ * @param {String} gql - GraphQL query from a gql template string
+ * @example gql`query { me { username } }`
+ * @param {Object} options
+ * @property {Object} user - Replace the default user or set it to null
+ * @returns {Any} Data matching the query
+ */
```

Comment:
@howard-e Any tips about improving the JSDoc?

Reply:
Sure!

1. So you typically use the `@property` directive when defining a `@typedef` or a namespace. So you'd use the `@property` directive to declare a property of that defined `@typedef`. I don't think that's what you want here with `user` so I think you could safely change that to using `@param` or create a `@typedef` for the `options` object and make it a property of that.
2. If you do choose to just make it a `@param` however, make sure you use `options.user` so the IDE could identify `user` as actually being a part of `options`
3. This is a nitpick, but usually, the pattern you'd see when representing primitives vs Classes/Types/Objects types is with the latter being your typical PascalCase (if that's how they're defined) and the former being lowercase which is what I think all of these types could be (primitives). All this isn't relevant of course if the project has already been using their own type standard when defining the JSDocs but could look into a best practice around it

Otherwise, it's looking good and descriptive enough. And I'm always a big fan of the `@example` being used so that's great!

Reply:
Ah I'm glad I asked, thanks for the help.

Here's what I ended up with:

```js
/**
 * Returns the data from a given query, useful for situations where errors are
 * not expected and no query variables are needed.
 * @param {GraphQLSyntaxTree} gql - GraphQL query from a gql template string
 * @example gql`query { me { username } }`
 * @param {object=} options
 * @param {object=} options.user - Replace the default user or set it to null
 * to simulate being logged out.
 * @returns {any} Data matching the query
 */
```

Reply:
Nice. Looks good!